test/Microsoft.ML.Tests/TrainerEstimators/MatrixFactorizationTests.cs

```diff
@@ -91,18 +98,79 @@ public void MatrixFactorizationSimpleTrainAndPredict()
                 var mlContext = new MLContext();
                 var metrices = mlContext.Regression.Evaluate(prediction, label: labelColumnName, score: scoreColumnName);
 
-                // Determine if the selected metric is reasonable for differen
-                var expectedWindowsL2Error = 0.61528733643754685; // Windows baseline
-                var expectedMacL2Error = 0.61192207960271; // Mac baseline
-                var expectedLinuxL2Error = 0.616821448679879; // Linux baseline
-                double tolerance = System.Math.Pow(10, -DigitsOfPrecision);
-                bool inWindowsRange = expectedWindowsL2Error - tolerance < metrices.L2 && metrices.L2 < expectedWindowsL2Error + tolerance;
-                bool inMacRange = expectedMacL2Error - tolerance < metrices.L2 && metrices.L2 < expectedMacL2Error + tolerance;
-                bool inLinuxRange = expectedLinuxL2Error - tolerance < metrices.L2 && metrices.L2 < expectedLinuxL2Error + tolerance;
-                Assert.True(inWindowsRange || inMacRange || inLinuxRange);
+                // Determine if the selected metric is reasonable for different platforms
+                double tolerance = Math.Pow(10, -7);
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
>Windows [](start = 63, length = 7)

you have to cases for OSPlatform.Windows, and this one says in comments this is Unix case, so can you change it to Unix?

Reply:
Fixed.

---
In reply to: [229016242](https://github.com/dotnet/machinelearning/pull/1407#discussion_r229016242) [](ancestors = 229016242)